packet_buffer.go

```diff
@@ -75,6 +82,25 @@ func autoDetectPacketSize(r io.Reader) (packetSize int, err error) {
 	return
 }
 
+// bufio.Reader can't be rewinded, which leads to packet loss on packet size autodetection
+// but it has handy Peek() method
+// so what we do here is peeking bytes for bufio.Reader and falling back to rewinding/syncing for all other readers
+func read(r io.Reader, b []byte) (shouldRewind bool, err error) {
```

Comment:
I think you should this function to `peek` directly

Reply:
I don't get the comment - `Peek()` is only available for `bufio.Reader`, I can't use it directly for any `io.Reader`

Reply:
Sorry I meant renaming the function to `peek`. I feel it's more accurate

Reply:
right, you got it